ngraph/core/src/op/loop.cpp

```diff
@@ -297,14 +301,25 @@ void op::v5::Loop::validate_and_infer_types()
             }
         }
     }
+
+    NODE_VALIDATION_CHECK(this,
+                          get_output_size() == m_output_descriptions.size(),
+                          "Number of outputs must be the same as number of output descriptions");
 }
 
 std::shared_ptr<Node> op::v5::Loop::clone_with_new_inputs(const OutputVector& new_args) const
 {
     NGRAPH_OP_SCOPE(v5_Loop_clone_with_new_inputs);
+    // WA: first input description with index 0 or 1 means that Loop consructor will duplicate it in
+    // the inputs.
+    // When using visit_attributes() no duplication occurs, input_offset shall be decremented.
+    size_t input_offset = 2;
+    if (m_input_descriptions[0].get()->m_input_index == 0 ||
+        m_input_descriptions[0].get()->m_input_index == 1)
+        input_offset--;
     // 0 - trip_count, 1 - execution condition, these inputs are not connected to the body
     // params
-    OutputVector body_params_args(new_args.begin() + 2, new_args.end());
+    OutputVector body_params_args(new_args.begin() + input_offset, new_args.end());
     auto op = make_shared<op::v5::Loop>(new_args[0], new_args[1]);
```

Comment:
I think we should add validation check on the beginning of the function.
e.g.: `NGRAPH_CHECK(new_args.size >= 2);`
Here index `0` and `1` without any check.

Reply:
This is `clone_with_new_inputs()`, it means that input args were already validated in `validate_and_infer_types()`.

Reply:
OK